gcovr/tests/test_args.py

```diff
@@ -150,6 +159,53 @@ def test_non_existing_directory_2_csv(capsys):
     helper_test_non_existing_directory_2_output(capsys, '--csv')
 
 
+def helper_test_non_writable_directory_output(capsys, option):
+    c = capture(capsys, [option, '/file.txt'])
```

Comment:
Hmm, this test is correct within the Docker container, but might have surprising results on a real system. Can we somehow guard this to only execute in an expected environment?

Reply:
We can run this check only inside docker. You mean the effect if someone develops gcov with an account having access to `/`? In windows we can use the environment `SystemRoot` which should also be protected for normal users.
Maybe we can try to create a file in the exepected directory and if this fails, we run the real test, else we skip the test at runtime.

Reply:
Yes, but it's a matter of principle: don't touch random files, only files in the current directory or in a tempdir. Even though no sane person should run tests as root or Admin.

My preferred solution would be to make sure that this test won't execute outside of the testing container, e.g. by setting an env variable in the container and checking for its existence in the test. E.g. `ENV GCOVR_ISOLATED_TEST=zkQEVaBpXF1i` with some randomly selected but static value to prevent false positives.

Reply:
Ok. I'll do it this way. The test is skipped if the environment variable has another value.

Reply:
I took your random ID and it's working.